src/libmscore/scorefile.cpp

```diff
@@ -457,7 +457,7 @@ bool MasterScore::saveFile(bool generateBackup)
             SetFileAttributes((LPCTSTR)backupDirNativePath.toLocal8Bit(), FILE_ATTRIBUTE_HIDDEN);
    #endif
 #else
-            SetFileAttributes((LPCTSTR)backupDirNativePath.toLocal8Bit(), FILE_ATTRIBUTE_HIDDEN);
+            SetFileAttributes((LPCTSTR)backupDirNativePath.toLocal8Bit().constData(), FILE_ATTRIBUTE_HIDDEN);
```

Comment:
replacing `toLocal8Bit().constData()` with `unicode()` works too, which means we could just drop the entire `#if (defined (_MSCVER) || defined (_MSC_VER))...#else...#endif` stuff, simplifying that preprocessor "mess" quite a bit?

Reply:
Actually the `toLocal8Bit().constData()` is needed in the above `#else` branche too anyway., So it should be:

   #ifdef Q_OS_WIN
            const QString backupDirNativePath = QDir::toNativeSeparators(backupDirString);
       #if (defined (UNICODE))
            SetFileAttributes((LPCTSTR)backupDirNativePath.unicode(), FILE_ATTRIBUTE_HIDDEN);
       #else
            // Use byte-based Windows function
            SetFileAttributes((LPCTSTR)backupDirNativePath.toLocal8Bit().constData(), FILE_ATTRIBUTE_HIDDEN);
       #endif
   #endif

Reply:
I think we can assume that Unicode is always on. So, we can simply do:
```
#ifdef Q_OS_WIN
        const QString backupDirNativePath = QDir::toNativeSeparators(backupDirString);
        SetFileAttributes((LPCTSTR)backupDirNativePath.unicode()->unicode(), FILE_ATTRIBUTE_HIDDEN);
#endif
```

Reply:
Works (compiles and links and runs without errors) in MinGW and MSVC, I checked both, with and without UNICODE being #define'd and even in 3.x (which doesn't `#define UNICODE` for MinGW at all, so I'm not sure whether it should be done here, that then would require a different fix for the other error though.)

Reply:
Well, there are several places in the code that switch UNICODE off explicitly, to get the char based wWindows functions.

IIRC the UNICODE thing was causing https://musescore.org/en/node/287955 ?

Reply:
As mentioned, in 3.x MinGW does not use/set UNICODE, I guess for a reason?

Reply:
@igorkorsukov, is the double unicode (`unicode()->unicode()`) in your [last comment](https://github.com/musescore/MuseScore/pull/7744#discussion_r596616234) intentional?

What's the consensus on whether the `#else` clause is still needed?

Reply:
> @igorkorsukov, is the double unicode (unicode()->unicode()) in your last comment intentional?

Yes, intentionally, I think it is more correct.
```
const QChar *QString::unicode() const
```
Cast `const QChar *`  to `LPCTSTR` works only on the basis of the internal implementation of `QChar`

Reply:
As far as I can tell the entire `#else` clause is not needed.

Reply:
> What's the consensus on whether the #else clause is still needed?

I see no reason for it, I believe that we can assume that `UNICODE` is always on by default.
If we need to turn it off somewhere, then turn it off only in these special places.

Reply:
I mean that matching the `#if (defined (_MSCVER) || defined (_MSC_VER))`. I'd rather keep the `UNICODE` part, for now, we may revisit that later, but as said, some (Windows related) partx in tthe code explicitly disable `UNICODE`

Reply:
> as said, some (Windows related) parts in the code explicitly disable UNICODE

Let those specific parts turn off Unicode if they need to (by way, PortMidi, propper not used in the MU4)
But why do we need to build the entire project with unicode disabled?
For example, in this case, if we disable unicode, it will stop working with any not ANSI (English) paths and file names.